cmd/tg_bot/internal/environment.go

```diff
@@ -154,3 +162,124 @@ func (tgEnv *TelegramBotEnvironment) NodesListMessage(urls []string) (string, er
 	}
 	return w.String(), nil
 }
+
+func (tgEnv *TelegramBotEnvironment) SubscribeToAllAlerts() error {
+	for alertType := range entities.AlertTypes {
+		alertName, ok := entities.AlertTypes[alertType] // check if such an alert exists
+		if !ok {
+			return errors.New("failed to subscribe to alerts, unknown alert type")
+		}
+
+		err := tgEnv.pubSubSocket.SetOption(mangos.OptionSubscribe, []byte{byte(alertType)})
+		if err != nil {
+			return err
+		}
+		tgEnv.subscriptions[alertType] = alertName
+		log.Printf("Subscribed to %s", alertName)
+	}
+
+	return nil
+}
+
+func (tgEnv *TelegramBotEnvironment) SubscribeToAlert(alertType entities.AlertType) error {
+	alertName, ok := entities.AlertTypes[alertType] // check if such an alert exists
+	if !ok {
+		return errors.New("failed to subscribe to alert, unkown alert type")
+	}
+
+	err := tgEnv.pubSubSocket.SetOption(mangos.OptionSubscribe, []byte{byte(alertType)})
+	if err != nil {
+		return errors.Wrap(err, "failed to subscribe to alert")
+	}
```

Comment:
What if you already has been subscribed to this alert?

Reply:
In handlers, it is checked that bot is not subscribed to the alert `cmd/tg_bot/internal/handlers/text.go:103`